### DISCUSSION TITLE
Generate Receipts from paid invoices

### DISCUSSION THREAD

**Discussion by Cata-ExF** [kimai/kimai#2745] [General]
Hi,

Is it possible to create a plugin that allows to generate a receipt for a selected invoice?
Almost all info for a receipt is in database: invoice number, customer, invoiced amount, even the date when invoice was marked as paid, only a transaction payment reference is missing.

When we mark an invoice as paid, the payment date popup is displayed, looks like a good place to add another optional field for transaction reference and a Download Receipt button. 

Thank you

Catalin

**Comment by Cata-ExF** [comment]
Hi,
Is the receipt feature planned for a future release?

I have another question: Is it possible to add a time entry with a negative amount in Fixed Rate field? For the moment, it accepts only positive values. If you can indicate please where the field validation settings are stored, will be very useful. For some clients, I charge a fixed upfront amount, then I need to deduct from the tracked time the hours paid in advance. Maybe it is possible to add manual items in invoice, as a discount?

Thank you

**Reply by kevinpapst** [reply]
Interesting question about the receipt. I don't know how that would look like, can you share a detailed idea/mockup/example PDF?

You cannot use negative values and I will not support bastardizing the field for a purpose that [it was never meant for](https://github.com/kevinpapst/kimai2/blob/master/src/Entity/Timesheet.php#L239).

But what is possible indeed: the invoice system supports arbitrary sources for line items (the Expense plugin uses this API) and we could work on a very small plugin that handles customer balances. I would have to test whether the invoice system can handle negative prices.
I am just not sure if a proper invoice (might depend on the jurisdiction where you live) can hold negative invoice items. But that's probably another topic...

**Comment by Cata-ExF** [comment]
Here are some examples of payment receipts:
https://support.google.com/adsense/thread/21572929/no-payment-number-on-payment-receipt?hl=en
https://www.receipts-templates.com/payment-receipt-template-free/
Or, can be a simple message, like: https://i2.wp.com/semioffice.com/wp-content/uploads/2021/08/Letter-of-Acknowledgment-of-Cheque-or-Payment-Received.png?ssl=1
A payment can be received in multiple ways: cash, card, bank transfers.

I believe it can be perfectly incorporated into the „mark as payed“ workflow, when we mark an invoice as paid.
